src/user/user.service.ts

```diff
@@ -0,0 +1,44 @@
+import { Injectable } from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import { Model } from 'mongoose';
+import { Role } from 'src/shared/enums/user-role.enum';
+import { CreateUserDto } from './dto/create-user.dto';
+import { UpdateUserDto } from './dto/update-user.dto';
+import { User, UserDocument } from './schemas/user.schema';
+
+@Injectable()
+export class UserService {
+  constructor(@InjectModel(User.name) private userModel: Model<UserDocument>) {}
+  
+  async create(createUserDto: CreateUserDto) {
+    const user = new this.userModel(createUserDto);
+    
+    return user.save();
+  }
+
+  async findAll() {
+    return this.userModel.find().exec();
+  }
+
+  async findOne(options: Partial<Omit<User, 'password'>>) {
+    return this.userModel.findOne(options).exec();
+  }
+
+  async findByRole(role: Role) {
+    return this.findOne({ role });
+  }
+
+  async findById(id: string) {
```

Comment:
Why `id` isn't included in `UserDTO` model?

Reply:
It is included in User schema – it’s missing in UpdateUserDto. Also, I don’t think that there should be a separate dto for just one string.

Reply:
Nobody said anything about creating a separate `DTO`. The question was only why `id` isn't included in `UserDTO` model. I don't know where you came to that conclusion

Reply:
There isn't such a thing as `UserDTO`. Do you mean `CreateUserDto` or `UpdateUserDto`?

Reply:
My bad. Had `UpdateUserDto` on my mind. Still, the conversation has been resolved 😄 so it doesn't matter anymore